source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -325,15 +242,7 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
     bool is_utf_16 = objectAddr & (1ULL << 60);
     uint64_t payloadAddr = objectAddr & ((1ULL << 56) - 1);
 
-    // FIXME: Remove printf debugging.
-    // stream.Printf("payload address = %llu\n", payloadAddr);
-    // stream.Printf("t (is-a-value) = %d\n", is_a_value);
-    // stream.Printf("v (sub-variant) = %d\n", is_Cocoa_or_small);
-    // stream.Printf("o (opaque) = %d\n", is_opaque);
-    // stream.Printf("w (width) = %d\n", is_utf_16);
-
-    if (!is_opaque && ((is_a_value && !is_Cocoa_or_small) ||
```

Comment:
I didn't understand the whole check to begin with.
I trust you that you can simplify it, but, maybe, can we add a comment?

Reply:
This checks for strings which aren't opaque or bridged. There is a "monster" comment above which documents why this class of strings need specific handling.

Reply:
But it could use a comment like "Handle plain-old native Swift strings"